Add E2E Tests to Validate File Integrity After Changes

KindImagination opened the pull request:
This PR introduces basic end-to-end (E2E) tests that check whether files remain intact and functional after modifications. The primary goal is to prevent accidental changes from affecting the integrity of critical test files.  

Additionally, this PR closes [Issue #42](https://github.com/razbakov/wedance/issues/42), which ensures automated checks are in place for validating the correctness of modified files.  

#### **Changes Introduced**:  
- Added E2E tests for verifying file integrity.  
- Ensured that all relevant test files remain consistent post-modifications.  
- Included step definitions for test scenarios related to file validation.  

#### **How to Test**:  
1. Go to the `e2e` folder and run `pnpm test` command.  
2. Run the E2E test suite using the configured testing framework.  
3. Confirm that all tests pass without failures.  
4. Modify test files and rerun the suite to validate expected behavior.

KindImagination committed:
feat(e2e): add basic page load tests for main sections

razbakov commented on e2e/features/artists.feature:
```diff
@@ -0,0 +1,8 @@
+
+  Scenario: Visit artists page
+    When I visit the artists page
+    Then I should see the artists container 
```
you shouldn't use jargon, Aina would never say "artists container"

razbakov commented on e2e/features/step_definitions/navigation.steps.js:
```diff
@@ -0,0 +1,37 @@
+  await this.cleanup()
+})
+
+Then('I should be on the courses page', async function () {
```
navigation feature doesn't have this, so no point having those steps here

KindImagination replied:
the steps are changed

razbakov commented on e2e/features/courses.feature:
```diff
@@ -0,0 +1,8 @@
+
+  Scenario: Visit courses page
+    When I visit the courses page
+    Then I should see the courses container 
```
each file should end with empty line

razbakov commented on e2e/features/step_definitions/groups.steps.js:
```diff
@@ -0,0 +1,14 @@
+})
+
+Then('I should see the groups container', async function () {
+  const container = await this.page.locator('main')
```
As we discussed the test should check that the following is present: header, filter and the first card.

This test will be successful on this broken page:

<img width="1014" alt="Screenshot 2025-02-11 at 00 20 50" src="https://github.com/user-attachments/assets/a7a6424b-5858-47e0-bab9-120bcc12d8cb" />

KindImagination replied:
Additional checks are included

KindImagination committed:
test(e2e): simplify e2e tests and fix navigation steps

- Simplify artists page test to check only essential elements
- Remove redundant checks from navigation steps
- Keep groups page test failing as expected (page not implemented)
- Clean up feature files and step definitions

KindImagination committed:
test(e2e): improve test readability and structure

- Replace technical jargon with user-friendly language in feature files
- Split group page tests into clear, descriptive steps
- Update step definitions to match new feature descriptions
- Add more detailed test scenarios for UI elements

KindImagination committed:
Merge conflict resolution

razbakov commented on components/VenuesView.vue:
```diff
@@ -293,8 +293,7 @@ const clearLocationFilter = () => {
           @update:location="
-              selectedLocation = $event;
-              showLocationFilter = false
+            ((selectedLocation = $event), (showLocationFilter = false))
```
what is this change for?

KindImagination replied:
rolled back

razbakov commented on composables/useGroup.ts:
```diff
@@ -1,15 +1,17 @@
     groupId,
   }
-}
+}
```
each file should have a new empty line

KindImagination replied:
empty line added

razbakov commented on e2e/features/groups.feature:
```diff
@@ -0,0 +1,10 @@
+  Scenario: Visit groups page
+    When I visit the groups page
+    Then I should see the page title
+    And I should see the search bar
```
not search bar, the filter

KindImagination replied:
renamed

KindImagination committed:
test(e2e): rename 'search bar' to 'filter' in groups feature

KindImagination committed:
bugfix

KindImagination committed:
prettier fixes

KindImagination committed:
prettier fix

KindImagination committed:
build error fix
